### PR TITLE
feat(batch): improve the implementation of BatchTaskMetrics

### LINKED ISSUE #5743: BatchTaskMetrics uses unbounded memory
### Is your feature request related to a problem? Please describe.

In #3832 and #4844, we have introduced task level metrics. To remove the metrics of a finished task, we create some new `Collector` for each new task, and `unregister` them when the task finishes. However, `prometheus::Registry::unregister` does not remove the data completely from the register - the `fq_name` (composed of the `Collector`'s help message and labels) remains in the registry.

https://github.com/tikv/rust-prometheus/blob/fc591fdee0ffc75cd26d8a7f82f8e2c4c66bb1ff/src/registry.rs#L93-L117

When the number of tasks is large, the `Collectors` can consume a lot of memory. (#5724) 

### Describe the solution you'd like

All the tasks should share the same `Collector`s, and distinguish themselves with variables label values. To avoid accumulating data from consuming too much memory, stale metrics should be cleaned by calling `remove_label_values` or `reset` on the `Collector`'s that has been set by a finished task. More specifically, we should clear all the metrics once they are collected.

### Describe alternatives you've considered

We might also maintain the metrics and labels modified by each task. However, there are several difficulties. 

- We do not know how many times a metric is modified.
- We do not know whether they are modified with the same label values. 
- The metrics may be of different types, so it's hard to maintain an `id -> metric` mapping. 

To achieve the deduplication of label values and the compatibility of different metric types, the implementation would be very complex.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gun9niR**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Instead of creating new `Collector`s for each task and put them in a queue for deletion, make all tasks share the same set of `Collector`s and clear collected data immediately.

**However, if data is inserted between `collect` and `reset`, it will be lost. It happens very rarely (2/100,000), and I fear introducing lock could block the collector thread for too long.**

### Performance

Insert 350,000 rows into a table (1 row per query): 

**this PR, cost 3 min** (please pay attention to the first 10 mins. I ran some heavy queries on prometheus later, resulting in the memory surge after 22:35.
this-pr

**main, cost 4.5 min**
main

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

Closes #5743

### CODE REVIEW

File: src/batch/src/executor/monitor/stats.rs [hunk 1]
```diff
@@ -125,144 +50,155 @@ macro_rules! def_task_metrics {
...
         }
+        for_all_task_metrics!(collect_and_clear);
+        mfs
     }
 }
 
-pub struct BatchMetrics {}
+/// A wrapper with `BatchTaskMetrics` and label values derived from the task id.
+#[derive(Clone)]
+pub struct BatchTaskMetricsWithLabels {
```
Comment (Gun9niR) [inline src/batch/src/executor/monitor/stats.rs:188]: I think the naming is a bit weird. 🤣 Any better ideas?

**Comment by ZENOTME** [comment]
Why does memory surge when running some heavy queries on Prometheus?

**Comment by Gun9niR** [comment]
> Why does memory surge when running some heavy queries on Prometheus?

"memory surge" is on Prometheus

**Review by ZENOTME — APPROVED** [review]
Cool! LGTM

**Review by BowenXiao1999 — APPROVED** [review]

**Pull request merged**